tests/test_20_url_tools.py

```diff
@@ -0,0 +1,45 @@
+import os
+
+import pytest
+
+from cads_adaptors.tools import url_tools
+
+
+@pytest.mark.parametrize(
+    "urls,expected_nfiles",
+    (
+        (
+            [
+                "https://get.ecmwf.int/repository/test-data/earthkit-data/test-data/test_single.nc"
+            ],
+            1,
+        ),
+        (
+            [
+                "https://get.ecmwf.int/repository/test-data/earthkit-data/test-data/test_single.nc",
+                "https://get.ecmwf.int/repository/test-data/earthkit-data/test-data/test_single.grib",
+            ],
+            2,
+        ),
+    ),
+)
```

Comment:
A couple of comments about this parametrization:
1. The internal logic of url_tools does not make any difference between len(1) or len(N>1) urls.
2. Why testing just the length of the output. You know exactly the output expected right?

Reply:
It was just to ensure that the handling of single files and multiple files was preserved.
Yes more can be added, but this is better than nothing

Reply:
In case it wasn't clear, I wasn't suggesting to add more tests. Just this:
```python
@pytest.mark.parametrize(
    "urls,expected_files",
    (
        (
            [
                "https://get.ecmwf.int/repository/test-data/earthkit-data/test-data/test_single.nc"
            ],
            ["repository/test-data/earthkit-data/test-data/test_single.nc"],
        ),
        (
            [
                "https://get.ecmwf.int/repository/test-data/earthkit-data/test-data/test_single.nc",
                "https://get.ecmwf.int/repository/test-data/earthkit-data/test-data/test_single.grib",
            ],
            [
                "repository/test-data/earthkit-data/test-data/test_single.nc",
                "repository/test-data/earthkit-data/test-data/test_single.grib",
            ],
        ),
    ),
)
def test_downloaders(tmp_path, monkeypatch, urls, expected_files):
    monkeypatch.chdir(tmp_path)  # try_download generates files in the working dir
    assert url_tools.try_download(urls) == expected_files
```